src/routes/reference/basic-reactivity/create-resource.mdx

```diff
@@ -17,206 +17,300 @@ description: >-
-	//`refetching` is true when the fetcher is triggered by calling `refetch()`,
-	// or equal to the optional data passed: `refetch(info)`
+### Related Types
+
+```typescript
+type ResourceReturn<T, R = unknown> = [Resource<T>, ResourceActions<T, R>]
+
+type Resource<T> = {
+  (): T | undefined
+  state: "unresolved" | "pending" | "ready" | "refreshing" | "errored"
+  loading: boolean
+  error: any
+  latest: T | undefined
 }
 
-const [data, { mutate, refetch }] = createResource(getQuery, fetchData)
+type ResourceActions<T, R = unknown> = {
+  mutate: (value: T | undefined) => T | undefined
+  refetch: (info?: R) => Promise<T> | T | undefined
+}
+
+type ResourceSource<S> = S | false | null | undefined | (() => S | false | null | undefined)
+
+type ResourceFetcher<S, T, R = unknown> = (
+  source: S,
+  info: { value: T | undefined; refetching: R | boolean }
+) => T | Promise<T>
+
+interface ResourceOptions<T, S = unknown> {
+  initialValue?: T
+  name?: string
+  deferStream?: boolean
+  ssrLoadFrom?: "initial" | "server"
+  storage?: (init: T | undefined) => [Accessor<T | undefined>, Setter<T | undefined>]
+  onHydrated?: (k: S | undefined, info: { value: T | undefined }) => void
+}
+```
+
+## Parameters
+
+### `source`
+- **Type:** `ResourceSource<S>`
+- **Default:** `undefined`
+
+Reactive data source whose non-nullish and non-false values are passed to the fetcher. 
+When the source changes, the fetcher is automatically re-run.
+
+### `fetcher`
+- **Type:** `ResourceFetcher<S, T, R>`
+
+Function that receives the source value (or `true` if no source), the current resource info, and returns a value or Promise.
+
+### `options`
+- **Type:** `ResourceOptions<T, S>`
+- **Default:** `{}`
+
+Configuration options for the resource.
+
+## Options
+
+### `initialValue`
+- **Type:** `T`
+- **Default:** `undefined`
+
+Initial value for the resource. 
+When provided, the resource starts in "ready" state and the type excludes `undefined`.
+
+### `name`
+- **Type:** `string`
+- **Default:** `undefined`
+
+A name for debugging purposes in development mode.
+
+### `deferStream`
+- **Type:** `boolean`
+- **Default:** `false`
+
+Controls streaming behavior during server-side rendering.
+
+### `ssrLoadFrom`
+- **Type:** `"initial" | "server"`
+- **Default:** `"server"`
+
+Determines how the resource loads during SSR hydration.
+
+### `storage`
+- **Type:** `(init: T | undefined) => [Accessor<T | undefined>, Setter<T | undefined>]`
+- **Default:** `createSignal`
 
-// read value
-data()
+Custom storage function for the resource value, useful for persistence or custom state management.
 
-// check if loading
-data.loading
+### `onHydrated`
+- **Type:** `(k: S | undefined, info: { value: T | undefined }) => void`
+- **Default:** `undefined`
 
-// check if errored
-data.error
+Callback fired when the resource hydrates on the client side.
 
-// directly set value without creating promise
-mutate(optimisticValue)
+## Return Value
```

Comment:
Sorry for being nitpicky again, but most of the pages use sentence case for headings, while a lot of the more recent content uses title case. I'm a bit confused. I don't have a strong opinion about which we choose, but we should at least be consistent. Since most existing pages use sentence case, I'd lean toward sticking with that.